src/Entities/Pawn.cpp

```diff
@@ -463,6 +466,14 @@ void cPawn::HandleFalling(void)
 
 		m_bTouchGround = true;
 		m_LastGroundHeight = GetPosY();
+
+		// Farmland trampling. Mobs smaller than 0.512 cubic blocks won't trample (Java Edition's behavior)
+		if (GetWorld()->IsFarmlandTramplingEnabled() &&
+			(BlockAtFoot == E_BLOCK_FARMLAND) &&
+			(GetWidth() * GetHeight() >= 0.512))
```

Comment:
Just realised - this doesn't calculate the volume at all! It's the area (is that what was intended)?

Reply:
Well... no. Another thing I'll need to fix

Reply:
Fixed. I'm also wondering - shouldn't a new member variable for volume (m_Volume), or even specifically for trampling (m_bTramplesFarmland) be introduced? Or would it be just an useless micro-optimization?

I guess micro-optimization, because any difference would probably only be visible with huge farms, with mobs roaming freely over them... but it never hurts to wonder & ask

Reply:
I would suggest this would be a pointless micro-optimisation to store a derived value of height and width (and might even slow things down if memory usage is expensive)